spec/solver_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rspec'
+require_relative '../solver'
+
+RSpec.describe Solver do
+  describe '#factorial' do
+    it 'returns factorial of any num' do
+      solver = Solver.new
+      result = solver.factorial(5)
+      expect(result).to eq(120)
+    end
+  end
+
+  describe '#factorial' do
+    it 'returns factorial of any num' do
+      solver = Solver.new
+      result = solver.factorial(-1)
+      expect(result).to eq('Value must be greater than or equal to 1.')
+    end
+  end
```

Comment:
- [ ] You should also implement a special case - when the input is 0, the result should equal 1. :+1: